Changed code to adhere to standard guidlines and added support for customization of captcha by providing an easy to use API

DhavalKapil opened the pull request.

DhavalKapil committed:
Converted tabs to 4 spaces

DhavalKapil committed:
Made changes according to coding guidlines

Added spaces around operators, in for loop
Used opening brace on the same line for control structures
Fixed indentation
Adhered to 80 characters per line

DhavalKapil committed:
Added documentation comments and added another variable for bitmap

Added documentation comments for instance variables and functions
Added another instance variable $bitmap to store the 3 dimensional array instead of $ascii

abiusx commented:
Its all good (though I don't seem to understand what "standard" you're talking about, there are plenty). Just one thing, I'm not sure using another instance variable is a good choice for $ascii, because it consumes a lot of memory (comparatively) and this method was used to reduce it.

DhavalKapil commented:
I was talking about PSR standard. I opted for having another variable so that `$ascii` can contain ascii string(for what it was named for).
Although a better option would be to load $ascii from a file and use only $bitmap.

abiusx commented:
Another file is definitely a bad idea. This is supposed to be a single-file no dependency library. For that purpose, it does not require to adhere to PSR. I personally do not believe PSR is a good standard, but have no opposition to your changes!
Let me know when its final and I’ll merge.
Thanks
-A

> On Jan 24, 2015, at 12:57 PM, Dhaval Kapil notifications@github.com wrote:
> 
> I was talking about PSR standard. I opted for having another variable so that $ascii can contain ascii string(for what it was named for).
> Although a better option would be to load $ascii from a file and using only $bitmap.
> 
> —
> Reply to this email directly or view it on GitHub https://github.com/OWASP/PureCaptcha/pull/1#issuecomment-71329553.

DhavalKapil committed:
Added functionality to customize text, length of CAPTCHA

Moved $captcha and $length to instance variables and initialized default values in constructor.
Took care to allow only the selected characters

DhavalKapil committed:
Shifted spacing, degree and scale factors to instance variables with appropriate getters and setters
Fixed documentation comments

DhavalKapil committed:
Shifted noiseLevel to instance variable, made charWidth, charHeight, ASCII constant and removed bool distort

DhavalKapil committed:
Updated README.md to show usage of PureCaptcha
